pkg/chunk/cassandra/table_client.go

```diff
@@ -65,3 +60,17 @@ func (c *tableClient) DescribeTable(ctx context.Context, name string) (desc chun
 func (c *tableClient) UpdateTable(ctx context.Context, current, expected chunk.TableDesc) error {
 	return nil
 }
+
+func (c *tableClient) getCreateTableQuery(desc *chunk.TableDesc) (query string) {
+	query = fmt.Sprintf(`
+		CREATE TABLE IF NOT EXISTS %s (
+			hash text,
+			range blob,
+			value blob,
+			PRIMARY KEY (hash, range)
+		)`, desc.Name)
+	if c.cfg.TableWith != "" {
+		query = fmt.Sprintf("%s WITH %s", query, c.cfg.TableWith)
```

Comment:
IMO, joining a string like this for a query to Cassandra is dangerous. If the end goal is to change the compaction strategy, I think that would be better handled with a `CompactionConfig` struct that a parameterizes all of the possible compactions strategies into a config and generates the appropriate query string.

Reply:
@jtlisi @pracucci 

Thank you for review! I agree your opinion.
And there are some considerations.
Some of table options for Cassandra are related to duration, like this:

```
gc_grace_seconds = 864000
default_time_to_live = 0
tombstone_compaction_interval = 864000
max_sstable_age_days = 100
...
```

I thought table options like above are suitable for `time.Duration` type.
But some of names for table options indicate time unit.
I am not sure if `time.Duration` is suitable.
There is a way to rename the table options but I'm concerned about user confusion. 
I am curious about your opinion.

Reply:
Since this is specifically for Cassandra,I recommend using types that will make input correspond to Cassandra. That way users already familiar with the system will be able to directly copy config values over.

So in the cases you mentioned above I would use `int` values. Just make sure you document each option in the flag help text.

Reply:
In our use case, we are using ScyllaDB, which is compatible but contains many options that are not default in Cassandra. 

https://docs.scylladb.com/getting-started/ddl/
CTRL-F scylla_encryption_options

It is a bad idea to assume what flags the Cassandra server will support. In my opinion this should be made extensible and the current PR does that nicely. 

This is a niche flag that will only be used by those who fully understand the implications of changing the flag, better solved by documentation, and already have authorization on the Cassandra server through the configuration.

Reply:
@jtlisi 
In addition to @coelho, there may be another use cases, such as DataStax products.
Also, table options depend on the Cassandra version.
It is hard to find all options and consider all differences.
My goals have been just caching and compaction strategy.
But this issue seems to have more considerations than I think.
If you still think joining string is dangerous, it may need another way.

Reply:
I see your point, and I agree in some cases a separate string makes sense. @pracucci WDYT?